### PR TITLE
Support Orion model

### PR DESCRIPTION
**Pull request by dachengai**
Support Orion model:  https://huggingface.co/OrionStarAI/Orion-14B-Base

### CODE REVIEW

**Comment by chuanzhubin** [comment]
加油加油

**Comment by Peter-Devine** [comment]
Much appreciated for adding this!

**Comment by kir152** [comment]
thanks for adding this

File: vllm/model_executor/models/__init__.py [hunk 1]
```diff
@@ -37,7 +37,8 @@
     "QWenLMHeadModel": ("qwen", "QWenLMHeadModel"),
     "RWForCausalLM": ("falcon", "FalconForCausalLM"),
     "StableLMEpochForCausalLM": ("stablelm", "StablelmForCausalLM"),
-    "YiForCausalLM": ("yi", "YiForCausalLM")
+    "YiForCausalLM": ("yi", "YiForCausalLM"),
+    "OrionForCausalLM": ("orion", "OrionForCausalLM")
```
Comment (esmeetu) [inline vllm/model_executor/models/__init__.py:41]: Could you sort this by alphabetical order?

File: vllm/model_executor/models/orion.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+from vllm.model_executor.layers.sampler import Sampler
+from vllm.model_executor.layers.vocab_parallel_embedding import (
+    VocabParallelEmbedding, ParallelLMHead)
+from vllm.model_executor.parallel_utils.parallel_state import (
+    get_tensor_model_parallel_world_size)
+from vllm.model_executor.sampling_metadata import SamplingMetadata
+from vllm.model_executor.weight_utils import (default_weight_loader,
+                                              hf_model_weights_iterator)
+from vllm.sequence import SamplerOutput
+from vllm.transformers_utils.configs.orion import OrionConfig
```
Comment (esmeetu) [inline vllm/model_executor/models/orion.py:29]: I think there's no need to add a new config file and replace this with `PretainedConfig` in `transformers`. So you can remove that config file.

**Review by esmeetu — COMMENTED** [review]
Hi, @dachengai. left some comments. You can also add this model in readme file.

File: vllm/model_executor/models/__init__.py (hunk 1 above)
Comment (dachengai) [reply vllm/model_executor/models/__init__.py:41]: thanks ! I'll change it

File: vllm/model_executor/models/orion.py (hunk 2 above)
Comment (dachengai) [reply vllm/model_executor/models/orion.py:29]: thanks !  I'll remove it !

**Comment by HanNayeoniee** [comment]
Is this work still in progress?

**Comment by Tygors** [comment]
> Even this work hasn't been merged yet, I tested on `dachengai:orion` branch. It worked! Thanks for your contribution 😊

i wonder where dachengai:orion branch is? and how can i test this model with vllm

**Comment by HanNayeoniee** [comment]
> > Even this work hasn't been merged yet, I tested on `dachengai:orion` branch. It worked! Thanks for your contribution 😊
> 
> i wonder where dachengai:orion branch is? and how can i test this model with vllm

`dachengai:orion` branch is [here](https://github.com/dachengai/vllm/tree/orion)
, but i only added a few changes like model config and init shown in `Files changed` tab.

You can test this model by changing the model path or name to Orion!
You can download it from huggingface.

**Comment by DevXiaolan** [comment]
Much appreciated for adding this!

**Comment by Tygors** [comment]
> > > Even this work hasn't been merged yet, I tested on `dachengai:orion` branch. It worked! Thanks for your contribution 😊
> > 
> > 
> > i wonder where dachengai:orion branch is? and how can i test this model with vllm
> 
> `dachengai:orion` branch is [here](https://github.com/dachengai/vllm/tree/orion), but i only added a few changes like model config and init shown in `Files changed` tab.
> 
> You can test this model by changing the model path or name to Orion! and you can download from huggingface.

many thanks, it works well for me.

**Comment by june42** [comment]
Any updates?

**Comment by chuanzhubin** [comment]
Why has it taken so long and still can't be merged in?

**Comment by WoosukKwon** [comment]
@esmeetu I addressed your review and updated the PR. Thanks again for the review!

**Pull request merged**